test/stability/testMerge/testMerge.go

```diff
@@ -89,151 +121,350 @@ func main() {
 	target := flag.String("target", "", "Target host name from config file (mandatory for VM)")
 	flag.StringVar(&dpdkLogLevel, "dpdk", "--log-level=0", "Passes an arbitrary argument to dpdk EAL")
 	flag.Parse()
-	if err := executeTest(*configFile, *target, testScenario); err != nil {
-		fmt.Printf("fail: %+v\n", err)
+	flow.CheckFatal(initDPDK())
+	if useSimpleGenerate {
+		generatorType = generate
+	} else {
+		generatorType = fastGenerate
 	}
+	flow.CheckFatal(executeTest(*configFile, *target, testScenario, pipelineScenario(pipelineType), generatorType, addSegmAfterMerge))
 }
 
-func executeTest(configFile, target string, testScenario uint) error {
-	if testScenario > 3 || testScenario < 0 {
-		return errors.New("testScenario should be in interval [0, 3]")
-	}
-	// Init NFF-GO system
-	config := flow.Config{
-		DPDKArgs: []string{dpdkLogLevel},
-	}
-	if err := flow.SystemInit(&config); err != nil {
-		return err
+// getter - is a part of pipeline getting packets by receive or generate or fast generate
+type getScenario uint
+
+const (
+	generate getScenario = iota
+	fastGenerate
+	recv
+)
+
+func setGetter(scenario getScenario, inputSource uint) (finalFlow *flow.Flow, err error) {
+	switch scenario {
+	case recv:
+		finalFlow, err = flow.SetReceiver(uint16(inputSource))
+	case fastGenerate:
+		if inputSource == useGenerator1 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
+	default:
+		if inputSource == useGenerator1 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
 	}
-	stabilityCommon.InitCommonState(configFile, target)
+	return finalFlow, err
+}
 
-	fixMACAddrs1 = stabilityCommon.ModifyPacket[outport1].(func(*packet.Packet, flow.UserContext))
-	fixMACAddrs2 = stabilityCommon.ModifyPacket[outport2].(func(*packet.Packet, flow.UserContext))
-	fixMACAddrs = stabilityCommon.ModifyPacket[outport1].(func(*packet.Packet, flow.UserContext))
+func handlePackets(pkt *packet.Packet, context flow.UserContext) {
+}
+
+// different test cases
+type pipelineScenario uint
+
+const (
+	getGet pipelineScenario = iota
+	segmGet
+	multipleSegm
+	singleSegm
+	getCopy
+	segmCopy
+)
+
+// set copier and handler to change packets in copied flow for future statistics
+func setCopyWithHandler(inFlow *flow.Flow) (*flow.Flow, error) {
+	outFlow, err := flow.SetCopier(inFlow)
+	if err != nil {
+		return nil, err
+	}
+	err = flow.SetHandler(outFlow, modifyCopiedPackets, nil)
```

Comment:
When you modify packets this way you do the following:
1) Copy as copy
2) Handle as start of new segment
3) Merge handle's output
It is a different story when you merge segments. Here we would like to merge Copy. You use recvCopy1 and recvCopy2 only with "+recv1" which is strange. No PASS/FAIL conditions on them. You should honestly set Copy. Without any starting segment (without any handle after it). We don't need to know about copied flows. In the final proportion they should be "*2".